tasks/apt.yml

```diff
@@ -40,6 +41,8 @@
     - python3-pip
     - python3-dev
     - python-virtualenv
+    - python3-pyparsing
+    - python3-software-properties
```

Comment:
It's believe its supported by Ubuntu and Debian, but then again scaling it in the future would require to make changes. So your way would be a better approach.